xrm/xrm.d.ts

```diff
@@ -437,49 +445,7 @@ declare namespace Xrm
          */
         refreshRibbon(): void;
 
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {"ERROR"}   level       An error message.
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, false if it fails.
-         */
-        setFormNotification( message: string, level: "ERROR", uniqueId: string ): boolean;
-
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {"WARNING"} level       A warning message.
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, false if it fails.
-         */
-        setFormNotification( message: string, level: "WARNING", uniqueId: string ): boolean;
-
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {"INFO"}    level       An informational message.
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, false if it fails.
-         */
-        setFormNotification( message: string, level: "INFO", uniqueId: string ): boolean;
-
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {string}    level       The level, as either "ERROR", "WARNING", or "INFO".
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, otherwise false.
-         */
-        setFormNotification( message: string, level: string, uniqueId: string ): boolean;
+        setFormNotification(message: string, level: Page.ui.FormNotificationLevel, uniqueId: string ): boolean;
```

Comment:
I debated about that myself.  What you list actually wouldn't break, since you are defining level to be a string literal type of "ERROR", which would satisfy the Page.ui.FormNotificationLevel type.  

```
var level1: "ERROR"; // Works
var level2 = "ERROR"; // Doesn't Work
var level3: Page.ui.FormNotificationLevel =  "ERROR"; // Works

ui.setFormNotification("message", level2, "id");  /// string is not assignable toFormNotificationLevel

// - Or -
ui.setFormNotification("message", level2 as Page.ui.FormNotificationLevel, "id");
```

One option is more type safe but potentially more burdensome.  The other way is less type safe, but also less burdensome.

If the standard way is to split it, sounds good to me, I'll create the overloads.